pulsar-broker/src/main/java/com/yahoo/pulsar/broker/loadbalance/BrokerHostUsage.java

```diff
@@ -71,25 +91,152 @@ public BrokerHostUsage(PulsarService pulsar) {
+    public synchronized SystemResourceUsage getBrokerHostUsage() throws IOException {
+        double totalNicUsageTx = getTotalNicUsageTxKb();
+        double totalNicUsageRx = getTotalNicUsageRxKb();
+        double memUsage = getTotalMemUsage();
+        CpuStat cpuStat = getTotalCpuUsage();
+        SystemResourceUsage usage = new SystemResourceUsage();
+        long now = System.currentTimeMillis();
+        if (lastCollection == 0L) {
+            usage.setMemory(new ResourceUsage(memUsage, totalMemLimit));
+            usage.setBandwidthIn(new ResourceUsage(0d, totalNicLimit));
+            usage.setBandwidthOut(new ResourceUsage(0d, totalNicLimit));
+            usage.setCpu(new ResourceUsage(0d, totalCpuLimit));
+        } else {
+            double elapsedSeconds = (now - lastCollection) / 1000d;
+            Double nicUsageTx = (totalNicUsageTx - lastTotalNicUsageTx) / elapsedSeconds;
+            Double nicUsageRx = (totalNicUsageRx - lastTotalNicUsageRx) / elapsedSeconds;
+            Long cpuTimeDiff = cpuStat.getTotalTime() - lastCpuStat.getTotalTime();
+            Long cpuUsageDiff = cpuStat.getUsage() - lastCpuStat.getUsage();
+            Double cpuUsage = ((double) cpuUsageDiff / (double) cpuTimeDiff) * totalCpuLimit;
+            usage.setMemory(new ResourceUsage(memUsage, totalMemLimit));
+            usage.setBandwidthIn(new ResourceUsage(nicUsageRx, totalNicLimit));
+            usage.setBandwidthOut(new ResourceUsage(nicUsageTx, totalNicLimit));
+            usage.setCpu(new ResourceUsage(cpuUsage, totalCpuLimit));
+        }
+
+        lastTotalNicUsageTx = totalNicUsageTx;
+        lastTotalNicUsageRx = totalNicUsageRx;
+        lastCpuStat = cpuStat;
+        lastCollection = System.currentTimeMillis();
+        return usage;
+    }
+
+    private double getTotalCpuLimit() {
+        return (double) (100 * Runtime.getRuntime().availableProcessors());
+    }
+
+    private CpuStat getTotalCpuUsage() {
         try {
-            /**
-             * Spawns a python process and runs the usage exporter script. The script return the machine information in
-             * the json format.
-             */
-
-            int exitCode = CmdUtility.exec(writer, usageScriptPath, "--host-usage-check-interval",
-                    Integer.toString(hostUsageCheckInterval));
-            if (exitCode != 0) {
-                LOG.warn("Process exited with non-zero exit code - [{}], stderr - [{}] ", exitCode, writer.toString());
-                throw new IOException(writer.toString());
+            String[] words = Files.lines(Paths.get("/proc/stat"))
+                    .findFirst()
+                    .get().split("\\s+");
+            LongStream values = Arrays.stream(words)
+                    .filter(s -> !s.contains("cpu"))
+                    .mapToLong(Long::parseLong);
+
+            return new CpuStat(values.sum(), values.limit(3).sum());
+        } catch (IOException e) {
+            return new CpuStat(0L, 0L);
```

Comment:
We should not blindly ignore exception, otherwise it would be very difficult to discover any problem.